pgx-pg-sys/build.rs

```diff
@@ -392,23 +416,632 @@ fn impl_pg_node(
+                            walker_fn(item, context);
+                            Node::traverse::<T>( unsafe { &mut *(item as *mut Node) }, walker_fn, context);
+                        }
+                    }
+                }
+            }
+            impl std::fmt::Display for List {
+                fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+                    write!(f, "[")?;
+                    for index in 0..self.length {
+                        if index != 0 {
+                            write!(f, ", ")?;
+                        }
+                        let item = unsafe { (*self.elements.offset(index as isize)).ptr_value as *mut Node };
+                        if item.is_null() {
+                            write!(f, "(null)")?;
+                        } else {
+                            write!(f, "{}", unsafe { *item })?;
+                        }
+                    }
+                    write!(f, "]")
+                }
+            }
+        }
+    };
 
+    pgnode_impls.extend(quote! {
+        unsafe fn list_length(list: *mut List) -> i32 {
+            if list.is_null() {
+                0
+            } else {
+                (*list).length
+            }
+        }
+        #list_fns
+
+        impl pg_sys::PgNode for Node {
+            fn traverse<T>(&mut self, walker_fn: fn(*mut Node, &mut T) -> (), context: &mut T) {
+                match self.type_ {
+                    #node_traverse_body
+                    _ => {},
+                }
+            }
+        }
+        impl std::fmt::Display for Node {
+            fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+                match self.type_ {
+                    #node_display_body
+                    _ => write!(f, "[unknown type {}]", self.type_),
+                }
+            }
+        }
+        impl pg_sys::PgNode for RangeTblEntry {
+            fn traverse<T>(&mut self, walker_fn: fn(*mut Node, &mut T) -> (), context: &mut T) {
+                // Derived from the implementation of `_outRangeTblEntry` in outfuncs.c.
+                // TODO: consider making the if/walk/traverse stanza into a macro?
+                if !self.alias.is_null() {
+                    walker_fn(self.alias as *mut Node, context);
+                    Node::traverse::<T>(unsafe { &mut *(self.alias as *mut Node) }, walker_fn, context);
+                }
+                if !self.eref.is_null() {
+                    walker_fn(self.eref as *mut Node, context);
+                    Node::traverse::<T>(unsafe { &mut *(self.eref as *mut Node) }, walker_fn, context);
+                }
+                // WRITE_ENUM_FIELD(rtekind, RTEKind);
+
+                match self.rtekind {
+                    RTEKind_RTE_RELATION => {
+                        // WRITE_OID_FIELD(relid);
+                        // WRITE_CHAR_FIELD(relkind);
+                        // WRITE_INT_FIELD(rellockmode);
```

Comment:
Fair enough, adding a note at the top that the stubbed/commented code sections are because this adheres fairly directly to the Postgres code layout might be useful in assuring they are preserved for posterity.